src/System.Console/src/Resources/Strings.resx

```diff
@@ -204,10 +204,16 @@
   <data name="InvalidOperation_PrintF" xml:space="preserve">
     <value>The printf operation failed.</value>
   </data>
+  <data name="InvalidOperation_ConsoleReadKeyOnFile" xml:space="preserve">
+    <value>Cannot read keys when either application does not have a console or when console input has been redirected from a file. Try Console.Read.</value>
+  </data>
```

Comment:
Sure. I don't think we need to be consistent with the desktop in the exception message string.